server.js

```diff
@@ -2988,6 +2988,67 @@ cache(function(data, match, sendBadge, request) {
   })}
 ));
 
+// example https://img.shields.io/mingle/https/your-company.mingle-api.thoughtworks.com/your-project-name/card-number/your-property-name.svg
+// Mingle card status integration
+camp.route(/^\/mingle\/(http|https)\/(.*)\/(.*)\/(.*)\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var scheme = match[1]; // http(s)
+  var host = match[2]; // your-company.mingle-api.thoughtworks.com
+  var project = match[3]; // your-project-name
+  var card = match[4]; // card-number e.g. `428`
+  var property = match[5]; // your-property-name e.g. `status`
+  var format = match[6];
+  var authToken = '';
+
+  var options = {
+    headers: {
+      'Content-Type': 'application/xml'
+    },
+    uri: scheme + '://' + host + '/api/v2/projects/' + project + '/cards/' + card + '.xml'
+  };
+
+  if (serverSecrets) {
+    // basic auth
+    options.headers.Authorization = 'Basic ' + new Buffer(serverSecrets.mingle_user + ':' + serverSecrets.mingle_password).toString('base64');
```

Comment:
What are the security restrictions put on the mingle password? I assume we're meant to keep it private, which prevents us from performing the request to http servers.